frontend/server/src/main/java/com/amazonaws/ml/mms/wlm/Model.java

```diff
@@ -116,19 +122,49 @@ public void addFirst(Job job) {
         jobsDb.get(DEFAULT_DATA_QUEUE).addFirst(job);
     }
 
-    public Job nextJob(String threadId) throws InterruptedException {
-        return nextJob(threadId, Long.MAX_VALUE);
-    }
+    public void pollBatch(String threadId, long waitTime, Map<String, Job> jobsRepo)
+            throws InterruptedException {
+        if (jobsRepo == null || threadId == null) {
```

Comment:
What if threadId is a blank string : "" ( a string of length 0) ? 
Can we use String.isEmpty() instead ?

Reply:
Sure. We could add it. 
Its technically not needed as the threadId can't be empty, and if it is, there wouldn't be a queue for it :) ..